crescent/internal/registry.py

```diff
@@ -302,22 +303,49 @@ def __build_commands(self) -> Sequence[AppCommand]:
 
         return tuple(built_commands.values())
 
-    async def __post_guild_commands(
-        self, commands: Sequence[CanBuild], guild: Snowflakeish
+    async def __post_application_commands(
+        self, commands: Sequence[AppCommand], guild: UndefinedOr[Snowflakeish]
     ) -> None:
         try:
             if self._application_id is None:
                 raise AttributeError("Client `application_id` is not defined")
+
+            existing_commands = await self._client.app.rest.fetch_application_commands(
+                application=self._application_id
+            )
+
+            if all(
+                any(
+                    map(
+                        lambda partial_command: command.is_partial_command(partial_command),
+                        existing_commands,
+                    )
+                )
+                for command in commands
+            ):
```

Comment:
Instead of `command.is_partial_command`, why can't we just use `__eq__`? That simplifies this code to `all(command in existing_commands for command in commands)`